pkg/contracts/test/ProxyOwnerTest.t.sol

```diff
@@ -0,0 +1,132 @@
+import {RegistryFactoryV0_1} from "../src/RegistryFactoryV0_1.sol";
+import {RegistryCommunityV0_0} from "../src/RegistryCommunityV0_0.sol";
+import {CVStrategyV0_0} from "../src/CVStrategyV0_0.sol";
+import {CollateralVault} from "../src/CollateralVault.sol";
+import {Upgrades} from "@openzeppelin/foundry/LegacyUpgrades.sol";
+import {ProxyOwner} from "../src/ProxyOwner.sol";
+import {ProxyOwnableUpgrader} from "../src/ProxyOwnableUpgrader.sol";
+
+contract ProxyOwnerTest is Test {
+    address deployerWallet = makeAddr("deployerWallet");
+    address anotherWallet = makeAddr("anotherWallet");
+    address protocolFeeReceiver = makeAddr("multisigReceiver");
+
+    function test_upgradeWithProxyAdmin() public {
+        vm.startPrank(deployerWallet);
+
+        ProxyOwner proxyOwner = ProxyOwner(
+            address(
+                new ERC1967Proxy(
+                    address(new ProxyOwner()), abi.encodeWithSelector(ProxyOwner.initialize.selector, deployerWallet)
+                )
+            )
+        );
+
+        ERC1967Proxy proxyRegistryFactory = new ERC1967Proxy(
+            address(new RegistryFactoryV0_0()),
+            abi.encodeWithSelector(
+                RegistryFactoryV0_0.initialize.selector,
+                address(proxyOwner), // owner
+                address(protocolFeeReceiver), // gardensFeeReceiver
+                address(new RegistryCommunityV0_0()), // registryCommunityTemplate
+                address(new CVStrategyV0_0()), // strategyTemplate
+                address(new CollateralVault()) // collateralVaultTemplate
+            )
+        );
+
+        Upgrades.upgradeProxy(
+            address(proxyRegistryFactory),
+            "RegistryFactoryV0_1.sol",
+            abi.encodeWithSelector(RegistryFactoryV0_1.initializeV2.selector, deployerWallet)
+        );
+    }
+
+    function test_upgradeWithEOAOwner() public {
+        vm.startPrank(deployerWallet);
+
+        ERC1967Proxy proxyRegistryFactory = new ERC1967Proxy(
+            address(new RegistryFactoryV0_0()),
+            abi.encodeWithSelector(
+                RegistryFactoryV0_0.initialize.selector,
+                address(deployerWallet), // owner
+                address(protocolFeeReceiver), // gardensFeeReceiver
+                address(new RegistryCommunityV0_0()), // registryCommunityTemplate
+                address(new CVStrategyV0_0()), // strategyTemplate
+                address(new CollateralVault()) // collateralVaultTemplate
+            )
+        );
+
+        Upgrades.upgradeProxy(
+            address(proxyRegistryFactory),
+            "RegistryFactoryV0_1.sol",
+            abi.encodeWithSelector(RegistryFactoryV0_1.initializeV2.selector, deployerWallet)
+        );
+    }
+
+    // function test_Revert_transferProxyAdminOwnership() public {
+    //     vm.startPrank(deployerWallet);
+
+    //     ProxyOwner proxyOwner = ProxyOwner(
+    //         address(
+    //             new ERC1967Proxy(
```

Comment:
missing test that revert and not allow normall address upgrade, that are critical tests to have. @kafann @Corantin

Reply:
https://github.com/1Hive/gardens-v2/pull/361#discussion_r1726171409